rkyv_test/src/test_std.rs

```diff
@@ -364,4 +364,28 @@ mod tests {
         value.insert(());
         test_archive(&value);
     }
+
+    // Test case for deriving attributes on an object containing an Optoin. The Option impls appear
+    // not to pass through attributes correctly.
+    #[test]
+    fn pass_thru_derive_with_option() {
```

Comment:
What functionality is this test exercising?

Reply:
That's a test for a bug which I found while working on this. Objects containing an Option did not get archived correctly.

It looks like the issue has since been fixed. The code that is the basis of this pull request is very old, and someone else appears to have found the same bug in the intervening time since all this code was written.

The test passes.